internal/gittuf-git/args/args.go

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package args
+
+import (
+	"strings"
+)
+
+/*
+	Argument processing for Git is a bit tricky, but we have the advantage of
+	only needing to support a few operations.
+
+	Significant inspiration drawn from
+	https://github.com/mislav/hub/blob/master/commands/args.go
+*/
+
+const (
+	noopFlag    = "--noop"
+	versionFlag = "--version"
+	listCmds    = "--list-cmds="
+	helpFlag    = "--help"
+	configFlag  = "-c"
+	chdirFlag   = "-C"
+	flagPrefix  = "-"
+)
+
+// Args stores any global flags for the invocation of Git (i.e. those before the
+// operation), the command/operation, and the parameters to said
+// command/operation.
+type Args struct {
+	GlobalFlags []string
+	Command     string
+	Parameters  []string
+	ConfigIdx   int
+	ChdirIdx    int
+}
+
+// ProcessArgs takes in the arguments from the commandline, and then nicely
+// organizes them into an Args struct. This should only be called with 1 or more
+// arguments, excluding the base one (i.e. the executable name)
+func ProcessArgs(args []string) Args {
+	if len(args) < 1 {
+		return Args{
+			GlobalFlags: []string{},
+			Command:     "",
+			Parameters:  []string{},
+		}
+	}
+
+	// Find the command's index
+	cmdIndex, configIndex, chdirIndex := locateCommand(args)
+
+	// Organize and return args
+	returnArgs := Args{
+		GlobalFlags: args[0:cmdIndex],
+		Command:     args[cmdIndex],
+		Parameters:  args[cmdIndex+1:],
+		ConfigIdx:   configIndex,
+		ChdirIdx:    chdirIndex,
+	}
+
+	return returnArgs
+}
+
+// locateCommand attempts to find where the main Git operation is in the
+// arguments. We need to do this as flags can be specified before the command.
+// In addition, it returns the location of any specified config directory and
+// change directory operation.
+// (The output format is (commandIndex, configIndex, chdirIndex))
+func locateCommand(args []string) (int, int, int) {
```

Comment:
This function is rather difficult to understand. Could you clarify what `skip` is, when we encounter it, etc.?